nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/test/java/org/nuxeo/ecm/platform/ec/notification/TestEmailNotification.java

```diff
@@ -0,0 +1,177 @@
+
+    protected static final String DOCUMENT_NAME = "anyFile";
+
+    protected static final String ANOTHER_DOCUMENT_NAME = "anyFile2";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected NotificationManager notificationManager;
+
+    @Inject
+    protected EventService eventService;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Inject
+    protected SmtpMailServerFeature.MailsResult emailsResult;
+
+    protected DocumentModel domain;
+
+    @Before
+    public void before() {
+        domain = session.createDocumentModel("/", "domain", "Domain");
+        domain = session.createDocument(domain);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToMainDocument() {
+        // create a File document and
+        DocumentModel documentModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification
+        addSubscriptions(documentModel);
+
+        // fire the dummy event
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), documentModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        // check the received mail
+        checkMailContent(documentModel, DOCUMENT_NAME);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToParentDocument() {
+        // create a simple hierarchy (domain <- DOCUMENT_NAME <- ANOTHER_DOCUMENT_NAME)
+        DocumentModel parentDocModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification on the parent document
+        addSubscriptions(parentDocModel);
+
+        // create a child document
+        DocumentModel mainDocModel = createDocument(parentDocModel, ANOTHER_DOCUMENT_NAME);
+
+        // fire the dummy event on the child document
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), mainDocModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        // check the received mail
+        checkMailContent(mainDocModel, ANOTHER_DOCUMENT_NAME);
+    }
+
+    protected void checkMailContent(final DocumentModel documentModel, String documentName) {
+        assertEquals(1, emailsResult.getMails().size());
+        SmtpMailServerFeature.MailMessage mailMessage = emailsResult.getMails().get(0);
+        assertEquals(String.format("[Dummy]Notification on the document '%s'", documentName), mailMessage.getSubject());
+
+        // check the mail content (see templates/dummyNotificationToSendMail.ftl)
+        String mailContent = getMailContent(mailMessage);
+        // description
+        assertTrue(mailContent.contains(String.format("The document %s is now available.", documentName)));
```

Comment:
Please write these booleans asserts with an additional `mailContent` in first argument as the message that will get displayed if the boolean is false, for diagnostic purposes.